server/recceiver/cfstore.py

```diff
@@ -355,460 +608,687 @@ def clean_service(self):
+def create_ioc_properties(
+    owner: str, iocTime: str, recceiverid: str, hostName: str, iocName: str, iocIP: str, iocid: str
+) -> List[CFProperty]:
+    """Create the properties from an IOC.
+
+    Args:
+        owner: The owner of the properties.
+        iocTime: The time of the properties.
+        recceiverid: The recceiver ID of the properties.
+        hostName: The host name of the properties.
+        iocName: The IOC name of the properties.
+        iocIP: The IOC IP of the properties.
+        iocid: The IOC ID of the properties.
+    """
     return [
-        create_property(owner, "hostName", hostName),
-        create_property(owner, "iocName", iocName),
-        create_property(owner, "iocid", iocid),
-        create_property(owner, "iocIP", iocIP),
-        create_active_property(owner),
-        create_time_property(owner, iocTime),
-        create_property(owner, RECCEIVERID_KEY, recceiverid),
+        CFProperty(CFPropertyName.hostName.name, owner, hostName),
+        CFProperty(CFPropertyName.iocName.name, owner, iocName),
+        CFProperty(CFPropertyName.iocid.name, owner, iocid),
+        CFProperty(CFPropertyName.iocIP.name, owner, iocIP),
+        CFProperty.active(owner),
+        CFProperty.time(owner, iocTime),
+        CFProperty(CFPropertyName.recceiverID.name, owner, recceiverid),
     ]
 
 
-def create_default_properties(owner, iocTime, recceiverid, channels_dict, iocs, cf_channel):
-    return create_properties(
-        owner,
-        iocTime,
+def create_default_properties(
+    ioc_info: IocInfo, recceiverid: str, channels_iocs: Dict[str, List[str]], iocs: Dict[str, IocInfo], cf_channel
+) -> List[CFProperty]:
+    """Create the default properties for an IOC.
+
+    Args:
+        ioc_info: The IOC information.
+        recceiverid: The recceiver ID of the properties.
+        channels_iocs: The dictionary of channel names to IOC IDs.
+        iocs: The dictionary of IOCs.
+        cf_channel: The Channelfinder channel.
+    """
+    channel_name = cf_channel.name
+    last_ioc_info = iocs[channels_iocs[channel_name][-1]]
+    return create_ioc_properties(
+        ioc_info.owner,
+        ioc_info.time,
         recceiverid,
-        iocs[channels_dict[cf_channel["name"]][-1]]["hostname"],
-        iocs[channels_dict[cf_channel["name"]][-1]]["iocname"],
-        iocs[channels_dict[cf_channel["name"]][-1]]["iocIP"],
-        channels_dict[cf_channel["name"]][-1],
+        last_ioc_info.hostname,
+        last_ioc_info.ioc_name,
+        last_ioc_info.ioc_IP,
+        last_ioc_info.ioc_id,
     )
 
 
 def __merge_property_lists(
-    newProperties: List[Dict[str, str]], channel: Dict[str, List[Dict[str, str]]], managed_properties: Set[str] = set()
-) -> List[Dict[str, str]]:
-    """
-    Merges two lists of properties ensuring that there are no 2 properties with
+    newProperties: List[CFProperty], channel: CFChannel, managed_properties: Set[str] = set()
```

Comment:
Might as well make this PEP compliant while you're it (-> `new_properties`) - I doubt anything calls it by kwarg